src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -212,6 +222,7 @@ public class NoOpClient : RemoteHostClient
             protected override Task<Session> TryCreateServiceSessionAsync(
                 string serviceName, PinnedRemotableDataScope snapshot, object callbackTarget, CancellationToken cancellationToken)
             {
+                snapshot?.Dispose();
```

Comment:
ah.. good finding. but rather than here, I would put it in the above method where you put try/catch